src/datagen/tsptwRealData.jl

```diff
@@ -0,0 +1,215 @@
+
+TsptwGeneratorFromRealData(n_city::Int, max_tw_gap::Int, max_tw::Int, pourcent_max_tw::Int64, file::String) = TsptwGeneratorFromRealData(n_city, max_tw_gap, max_tw, pourcent_max_tw, file, true)
+
+"""
+    fill_with_generator!(cpmodel::CPModel, gen::TsptwGeneratorFromRealData)::CPModel    
+Fill a CPModel with the variables and constraints generated. We fill it directly instead of 
+creating temporary files for efficiency purpose !
+
+Basicaly finds positions with a uniform distributions, then sets the time windows by creating a feasible tour and adding
+some randomness by using uniform distributions with gap and the length of the time windows.
+
+A seed must be specified by the user to generate a specific instance. As long as Random.seed!(seed) is called at the beginning of the function, every random-based operations with be deterministic. Caution : this is not the seed that must be specified in order to generate a same set of evaluation instances across experiment, in that case, the user must call Random.seed! only once, at the beginning of the experiment. 
+"""
+function fill_with_generator!(cpmodel::SeaPearl.CPModel, gen::TsptwGeneratorFromRealData; rng::Union{Nothing,AbstractRNG} = nothing, dist = nothing, timeWindows=nothing)
+
+    rng = isnothing(rng) ? MersenneTwister() : rng
+
+    lines = ""
+    open(gen.file, "r") do openedFile
+        input = read(openedFile, String)
+        lines = split(input, '\n')
+    end
+    max_city = parse(Int64, lines[1])
+    distance = zeros(Int64, max_city, max_city)
+    timeWindow = zeros(Int64, max_city, 2)
+    for i in 1:max_city
+        di = split(lines[i + 1], " ")
+        for j in 1:max_city
+            distance[i,j] = parse(Int64, di[j])
+        end
+    end
+
+    for i in 1:max_city
+        ti = split(lines[i + 1 + max_city], " ")
+        ti = filter!(e->e≠"",ti)
+        timeWindow[i,1] = parse(Int64, ti[1])
+        timeWindow[i,2] = parse(Int64, ti[2])
+    end
+    
+    perm  = shuffle(rng, 1:max_city)
+    randomCity = randperm!(rng, perm)[1:gen.n_city]
+    dist = zeros(Int64, gen.n_city, gen.n_city)
+    timeWindows = zeros(Int64, gen.n_city, 2)
+    maxTW = floor(Int, gen.pourcent_max_tw * gen.n_city / 100)
+    for i in 1:gen.n_city
+        for j in 1:gen.n_city
+            dist[i, j] = distance[randomCity[i], randomCity[j]]
+        end     
+    end
+    maxValue = sum(dist[i,j] for i in 1:gen.n_city, j in 1:gen.n_city)
+
+    for i in 1:maxTW
+        timeWindows[i, 1] = timeWindow[randomCity[i], 1]
+        timeWindows[i, 2] = timeWindow[randomCity[i], 2]
+    end
+    for i in maxTW+1:gen.n_city
+        timeWindows[i, 1] = 0
+        timeWindows[i, 2] = maxValue
+    end
+
+    random_solution = [1, shuffle(rng, Vector(2:gen.n_city))]
+
+    #TODO need to find a way to retrieve coordinates of points from 
+    #x_pos = zeros(gen.n_city)
+    #y_pos = zeros(gen.n_city)
+    #grid_size = 0
+    #cpmodel.adhocInfo = dist, timeWindows, hcat(x_pos, y_pos), grid_size
```

Comment:
It seems to me that in order to use the tsptw specific rewards, you have to fill in `adhocInfo`.

Since this new generator does not seem essential to me, maybe just remove it from the pull request until it is functional.

Reply:
That's right, I moved the file `tsptwRealData.jl` and its dependencies on the branch [Tsptw-Real-Data](https://github.com/corail-research/SeaPearl.jl/tree/Tsptw-Real-Data).